src/Storefront/Framework/Cache/ReverseProxy/RedisReverseProxyGateway.php

```diff
@@ -57,6 +57,8 @@ public function __construct(
     public function tag(array $tags, string $url, Response $response): void
     {
         foreach ($tags as $tag) {
+            // To avoid duplicates of the same url inside one tag
+            $this->redis->lRem($tag, $url, 1);
```

Comment:
You cannot be sure what is in that index. I think we need to switch to sets instead of lists

Reply:
LREM do not take number index as parameter. Thrid parameter is count - how many records we want to delete after this value.
![image](https://github.com/shopware/platform/assets/43290585/2d8f516c-5374-4190-9370-565a1ac7af3b)

Reply:
But we don't know how they are ordered there and don't know that the last item was ours. So I would say it would make more sense to switch over to sets and redis makes sure it's valid